extra_data/reader.py

```diff
@@ -1449,6 +1449,16 @@ def open_run(
         files which don't have this flag, out-of-sequence train IDs are suspect.
         If True, it tries to include these trains.
     """
+    if data is Ellipsis or data == 'all':
+        args = (proposal, run)
+        kwargs = dict(include=include, file_filter=file_filter,
+                      inc_suspect_trains=inc_suspect_trains)
+
+        dc = open_run(*args, data='raw', **kwargs).union(
+            open_run(*args, data='proc', **kwargs))
```

Comment:
I'm afraid you're testing that in a very specific case. In practice there are keys duplication between raw and proc data, which will be conflicting when merging the 2 data collections.

Reply:
Ouch, how ironic the very problem I'm trying to avoid with REMI is biting this solution now. I'll take a look whether one could add an `ignore_conflicts` flag to `DataCollection.union()`.

Reply:
You could use `raw_extra = raw.deselect(proc)`

Reply:
Just remembering, I think `.deselect()` works for this where it can drop entire files for the resulting DataCollection, as with the big detectors which are in their own files, but it won't work if it deselects some sources but still has other data from the same files, because `.union()` will effectively reselect these. I think this was part of why we postponed this before.

(I'm not sure I've explained that very clearly. I can have another go if it doesn't make sense)

Reply:
Yes, I've just noticed this. When calibrated data adds a virtual source, `.deselect()` will throw an exception.
But since we filter on the source level here anyway, I might as well exclude those explicitly.